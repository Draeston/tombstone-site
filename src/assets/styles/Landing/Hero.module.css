@import url('https://fonts.googleapis.com/css2?family=IBM+Plex+Sans+Arabic:wght@200;300;400&display=swap');

.hero {
    background: url("../../images/Landing/tombstone-hero.jpg") no-repeat;
    background-position-y: 80%;
    background-size: cover;
    position: relative;
    padding: 50px;

    min-height: 300px;
}

/* There I don't follow BEM, because it's inefficient */
.heroText {
    font-family: 'IBM Plex Sans Arabic', sans-serif;
    font-size: 24px;
    color: white;
    text-align: center;
}
