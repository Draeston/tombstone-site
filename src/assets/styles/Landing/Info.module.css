/*noinspection CssUnknownTarget*/
@import url("https://fonts.googleapis.com/css2?family=IBM+Plex+Sans+Arabic:wght@200;300;400&display=swap");

.info {
  background: white;
  position: relative;
  width: 80%;
  margin: 50px auto;
  max-height: 300px;
}

.infoText, .infoTextRight {
  font-family: "IBM Plex Sans Arabic", sans-serif;
  font-size: 24px;
  color: black;
  display: inline-block;
  width: 50%;
  position: relative;
  top: -120px;
}

.infoTextRight {
  text-align: right;
}

.imgWrapper {
  display: inline-block;
  width: 50%;
  text-align: center;
}

.tombImg {
  max-height: 300px;
}
