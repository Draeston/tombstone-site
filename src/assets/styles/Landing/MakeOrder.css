@import url("https://fonts.googleapis.com/css2?family=Creepster&display=swap");
@import url('https://fonts.googleapis.com/css2?family=Sancreek&display=swap');

.make-order {
  background: url("../../images/Landing/makeorder.jpg") no-repeat;
  background-size: cover;
  background-position-y: 80%;
  min-height: 300px;
  margin-top: 30px;
  padding-top: 15px;
  text-align: center;
  position: relative;
}

.make-order-title {
  font-family: "Creepster", cursive;
  font-size: 40px;
  color: white;
  text-align: center;
  letter-spacing: 5px;
}

.make-order-button {
  padding: 10px 20px;
  font-family: 'Sancreek', cursive;
  font-size: 24px;
  color: white;
  text-align: center;
  background: darkred;
  border-radius: 15px;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

a {
  text-decoration: none;
}
