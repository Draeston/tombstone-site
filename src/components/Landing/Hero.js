import styles from "../../assets/styles/Landing/Hero.module.css";

export function Hero() {
  return (
    <div className={styles.hero}>
      <h1 className={styles.heroTitle}>TOMBSTONES</h1>
      <p className={styles.heroText}>
        This web-site was designed to serve the purpose for fast decision
        process on gravestone selection
      </p>
    </div>
  );
}
