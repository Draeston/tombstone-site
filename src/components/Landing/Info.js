import styles from "../../assets/styles/Landing/Info.module.css";
import React from "react";

export function Info({ text, isReverted = true, image }) {
  return (
    <div className={styles.info}>
      {isReverted ? (
        <React.Fragment>
          <span className={styles.infoTextRight}>{text}</span>
          <div className={styles.imgWrapper}>
            <img
              className={styles.tombImg}
              src={require(`../../assets/images/Landing/Info/${image}`)}
              alt=""
            />
          </div>
        </React.Fragment>
      ) : (
        <React.Fragment>
          <div className={styles.imgWrapper}>
            <img
              className={styles.tombImg}
              src={require(`../../assets/images/Landing/Info/${image}`)}
              alt=""
            />
          </div>
          <span className={styles.infoText}>{text}</span>
        </React.Fragment>
      )}
    </div>
  );
}
