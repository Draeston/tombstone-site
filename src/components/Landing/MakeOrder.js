import "../../assets/styles/Landing/MakeOrder.css";
import { NavLink } from "react-router-dom";

function MakeOrder() {
  return (
    <div className="make-order">
      <h1 className="make-order-title">JOIN US!</h1>
      <NavLink to="/tombstones" className="make-order-button">
        List of tombstones
      </NavLink>
    </div>
  );
}

export default MakeOrder;
