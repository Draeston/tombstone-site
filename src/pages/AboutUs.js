import {Hero} from "../components/AboutUs/Hero";
import {Info} from "../components/AboutUs/Info";

export function AboutUs () {
    return (
        <div>
            <Hero />
            <Info />
        </div>
    )
}
