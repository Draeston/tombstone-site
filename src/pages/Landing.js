// background images will be changed later
import { Hero } from "../components/Landing/Hero";
import { Info } from "../components/Landing/Info";
import MakeOrder from "../components/Landing/MakeOrder";

export function Landing() {
  const infoText = [];
  infoText.push(
    "We are glad to help the customers to decide and design proper and cheap product that will delivered their waits as fast as possible and to avoid any difficulties that can occur."
  );
  infoText.push(
    "Our tombstones are really of high quality and we will make sure it will not be destroyed in 5-10 years"
  );
  infoText.push(
    "We will make sure that your buried will rest in the ground calmly"
  );
  return (
    <div>
      <Hero />
      <Info text={infoText[0]} image="1.png" isReverted={false} />
      <Info text={infoText[1]} image="2.png" />
      <Info text={infoText[2]} image="3.png" isReverted={false} />
      <MakeOrder />
    </div>
  );
}
