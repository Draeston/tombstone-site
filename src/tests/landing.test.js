import React from "react";
import renderer from "react-test-renderer";
import { Landing } from "../pages/Landing";
import { BrowserRouter } from "react-router-dom";

test("snapshot landing", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Landing />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
