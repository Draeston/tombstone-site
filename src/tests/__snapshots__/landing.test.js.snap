// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot landing 1`] = `
<div>
  <div
    className="hero"
  >
    <h1
      className="heroTitle"
    >
      TOMBSTONES
    </h1>
    <p
      className="heroText"
    >
      This web-site was designed to serve the purpose for fast decision process on gravestone selection
    </p>
  </div>
  <div
    className="info"
  >
    <div
      className="imgWrapper"
    >
      <img
        alt=""
        className="tombImg"
        src="1.png"
      />
    </div>
    <span
      className="infoText"
    >
      We are glad to help the customers to decide and design proper and cheap product that will delivered their waits as fast as possible and to avoid any difficulties that can occur.
    </span>
  </div>
  <div
    className="info"
  >
    <span
      className="infoTextRight"
    >
      Our tombstones are really of high quality and we will make sure it will not be destroyed in 5-10 years
    </span>
    <div
      className="imgWrapper"
    >
      <img
        alt=""
        className="tombImg"
        src="2.png"
      />
    </div>
  </div>
  <div
    className="info"
  >
    <div
      className="imgWrapper"
    >
      <img
        alt=""
        className="tombImg"
        src="3.png"
      />
    </div>
    <span
      className="infoText"
    >
      We will make sure that your buried will rest in the ground calmly
    </span>
  </div>
  <div
    className="make-order"
  >
    <h1
      className="make-order-title"
    >
      JOIN US!
    </h1>
    <a
      aria-current={null}
      className="make-order-button"
      href="/tombstones"
      onClick={[Function]}
    >
      List of tombstones
    </a>
  </div>
</div>
`;
